idea/idea-fir/src/org/jetbrains/kotlin/idea/quickfix/fixes/AddWhenRemainingBranchFixFactories.kt

```diff
@@ -0,0 +1,75 @@
+ */
+
+package org.jetbrains.kotlin.idea.quickfix.fixes
+
+import org.jetbrains.kotlin.diagnostics.WhenMissingCase
+import org.jetbrains.kotlin.idea.KotlinBundle
+import org.jetbrains.kotlin.idea.api.applicator.HLApplicator
+import org.jetbrains.kotlin.idea.api.applicator.HLApplicatorInput
+import org.jetbrains.kotlin.idea.api.applicator.applicator
+import org.jetbrains.kotlin.idea.fir.api.fixes.HLQuickFix
+import org.jetbrains.kotlin.idea.fir.api.fixes.diagnosticFixFactory
+import org.jetbrains.kotlin.idea.frontend.api.analyse
+import org.jetbrains.kotlin.idea.frontend.api.components.ShortenOption
+import org.jetbrains.kotlin.idea.frontend.api.fir.diagnostics.KtFirDiagnostic
+import org.jetbrains.kotlin.idea.frontend.api.symbols.KtClassKind
+import org.jetbrains.kotlin.idea.frontend.api.tokens.HackToForceAllowRunningAnalyzeOnEDT
+import org.jetbrains.kotlin.idea.frontend.api.tokens.hackyAllowRunningOnEdt
+import org.jetbrains.kotlin.idea.util.generateWhenBranches
+import org.jetbrains.kotlin.name.ClassId
+import org.jetbrains.kotlin.psi.KtWhenExpression
+
+object AddWhenRemainingBranchFixFactories {
+    class Input(val whenMissingCases: List<WhenMissingCase>, val enumToStarImport: ClassId?) : HLApplicatorInput
+
+    val applicator: HLApplicator<KtWhenExpression, Input> = getApplicator(false)
+    val applicatorUsingStarImport: HLApplicator<KtWhenExpression, Input> = getApplicator(true)
+
+    @OptIn(HackToForceAllowRunningAnalyzeOnEDT::class)
+    private fun getApplicator(useStarImport: Boolean = false) = applicator<KtWhenExpression, Input> {
+        familyAndActionName(
+            if (useStarImport) KotlinBundle.lazyMessage("fix.add.remaining.branches.with.star.import")
+            else KotlinBundle.lazyMessage("fix.add.remaining.branches")
+        )
+        applyTo { whenExpression, input ->
+            generateWhenBranches(whenExpression, input.whenMissingCases)
+            val shortenCommand = hackyAllowRunningOnEdt {
+                analyse(whenExpression) {
+                    collectPossibleReferenceShorteningsInElement(
+                        whenExpression,
+                        callableShortenOption = {
+                            if (useStarImport && it.callableIdIfNonLocal?.classId == input.enumToStarImport) {
+                                ShortenOption.SHORTEN_AND_IMPORT_ALL_IN_PARENT
+                            } else {
+                                ShortenOption.DO_NOT_SHORTEN
+                            }
+                        })
+
+                }
+            }
+            shortenCommand.invokeShortening()
+        }
+    }
+
+    @OptIn(ExperimentalStdlibApi::class)
+    val noElseInWhen = diagnosticFixFactory(KtFirDiagnostic.NoElseInWhen::class) { diagnostic ->
+        val whenExpression = diagnostic.psi
+        if (whenExpression.closeBrace == null) return@diagnosticFixFactory emptyList()
```

Comment:
Maybe we just can generate it? what if the code is following `when(expression)` -- we can generate braces in such case

Reply:
Yep, sounds good! Also added tests covering these cases.